core/include/scipp/core/transform.h

```diff
@@ -180,10 +180,18 @@ template <class T> struct ValuesAndVariances {
     throw std::runtime_error(
         "`begin` not implemented for sparse data with variances.");
   }
+  void *begin() const {
+    throw std::runtime_error(
+        "`begin` not implemented for sparse data with variances.");
+  }
   void *end() {
     throw std::runtime_error(
         "`end` not implemented for sparse data with variances.");
   }
+  void *end() const {
```

Comment:
Because it amounts to implementing a proxy iterator in the style of what `rangev3::zip` provides. This is quite a difficult task, I think. A better solution might thus be to provide a proxy object with more features such as `concatenate`. This it turn would require that `sparse_container` supports those operations as well, i.e., we cannot simply use a typedef for `boost::small_vector`. Implementing this should be relatively straight forward, but have not gotten around to do so.